crates/bevy_ecs/src/schedule/system_container.rs

```diff
@@ -118,6 +118,13 @@ pub struct ParallelSystemContainer {
 unsafe impl Send for ParallelSystemContainer {}
 unsafe impl Sync for ParallelSystemContainer {}
 
+impl Drop for ParallelSystemContainer {
+    fn drop(&mut self) {
+        // SAFE: pointer originally constructed by Box::into_raw
+        unsafe { Box::from_raw(self.system.as_mut()) };
```

Comment:
Why does it store `NonNull<_>` instead of `Box<_>` anyway?

Reply:
Because of `system_mut_unsafe`, which returns `&mut` ref out of `&self`. This is impossible with raw `Box`.

Reply:
Yep, `&T` -> `&mut T` is UB afaik.
You could maybe do something like `UnsafeCell<Box<_>>` but ehh.... :shrug:

Reply:
I think `Box<UnsafeCell<_>>` would be better, but `UnsafeCell<Box<_>>` is probably fine too.

Reply:
Switched to `Box<UnsafeCell<_>>` and got rid of custom Drop impl.